### DISCUSSION TITLE
Can we access the notebook's state from inside the notebook?

### DISCUSSION THREAD

**Discussion by markNZed** [fonsp/Pluto.jl#1458] [Questions about using Pluto]
I'm wondering if notebook metaprogramming is possible from within the same notebook e.g. generating multiple outputs by generating multiple cells. Are there any examples of this being done? Thanks.

It seems I can access the PlutoRunner variable from inside a notebook cell. Perhaps anothe way is to use the Javscript frontend to modify the notebook (I guess this is preferrable).  It would be great to see how Javasript embedded in a cell could be used to generate a new cell.

From Javascript console it seems there is editor_state available. I can do something like:
`editor_state.cell_inputs_local["6ea4b044-11bb-49ba-8f5c-54f4572925ce"].code = 'md"hello world"'`
which updates editor_state but does not update the UI.

**Comment by disberd** [comment]
It is technically possible but it is not straightforward at all.
There was some discussion on it on zulip starting from [this message](https://julialang.zulipchat.com/#narrow/stream/257776-Pluto.2Ejl-development/topic/Modify.20notebook.20appearance.20from.20personal.20package/near/243625540), but the main concept is that you should mostly work and send commands from the julia side, and not the javascript one.
The `editor_state` variable is mostly for reading and changes you put in there are not reflected on the actual notebook state that is hidden on the Preact classes of the notebook.

I also had another discussion about adding the capability of executing js in notebook programmatically here https://github.com/fonsp/Pluto.jl/discussions/1359, but the previously tagged discussion on zulip also had some arguments of why doing so is problematic.
